### ISSUE TITLE
Postman CollectionTypeError on empty examples object

### ISSUE THREAD

**Issue by miriamgreis** [apideck-libraries/portman#652]
If I use the following stripped-down OpenAPI file, I get an error during the conversion saying: `Converting OpenAPI to Postman Collection Type Error: Cannot read properties of undefined (reading '$ref')`.

The problem is the empty `examples: {}` which is sometimes autogenerated by Stoplight Studio. This wasn't a problem with earlier versions of Portman, so I assume that it is related to some recent openapi-format updates. I'm not sure whether this can be fixed here, but for me it's most important that Portman fails on such errors when I use it in the CI. Otherwise, nobody will realise that the tests are not generated at all.

```
openapi: 3.0.1
info:
  title: Test
  version: 1.0.0
  description: Test
servers:
  - url: https://example.de/test
security:
  - bearerToken: []
tags:
  - name: Test API
paths:
  /test:
    get:
      summary: Get test
      tags:
        - Test API
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  test-value:
                    type: string
              examples: {}
      operationId: get-test
      description: Test
  securitySchemes:
    bearerToken:
      type: http
      scheme: bearer
```

**Comment by thim81** [comment]
hi @miriamgreis 

Looking into the matter. 
Do you apply a filter on the OpenAPI file?

**Comment by miriamgreis** [comment]
No, just running Portman with the local OpenAPI file. Nothing special.

**Comment by thim81** [comment]
Openapi-format is only triggered with there is a filterFile, so I'll look further in the matter.

Do you happen to know, in which version it still worked?

**Comment by miriamgreis** [comment]
I just tried to figure it out as I had a very old version installed locally. The first version that I see the error message with is v1.29.0, but you might want to doublecheck.

**Comment by thim81** [comment]
@miriamgreis Found the issue: the openapi-to-postman "convertV2" fails. We introduced the V2 method, which has become the default by Postman, in [v1.29](https://github.com/apideck-libraries/portman/blob/main/CHANGELOG.md#v1290---2024-08-06).

I'm created an the [issue](https://github.com/postmanlabs/openapi-to-postman/issues/824) in the openapi-to-postman github.

**Comment by thim81** [comment]
It should fail:

```
postmanObj = await oaToPostman.convert(oaToPostmanConfig).catch(err => {
        console.error('Postman Collection generation failed: ', err.toString())
        process.exit(1)
      })
```

Let me double check if the error trigger is not properly catched in the nesting.

**Comment by thim81** [comment]
hi @miriamgreis, an update: We just submitted a [PR](https://github.com/postmanlabs/openapi-to-postman/pull/825) that should overcome the error in the openapi-to-postman library.

**Comment by thim81** [comment]
as possible work-arounds for now:
- manually remove the empty example
- If you would use any filter or only a filter with `preserveEmptyObjects: false`, it will sanitize the OpenAPI spec of empty objects.

Since openapi-to-postman does not throw an error and returns a postman obj, we cannot catch the error with proper exit 1 code. In the PR on openapi-to-postman, we added the throwing of an error.

**Comment by miriamgreis** [comment]
Perfect, thanks. For now I anyway manually removed the empty example anyway. I saw your PR for openapi-to-postman, so seems that everything needs to be resolved on their side.

**Comment by thim81** [comment]
update: The [PR](https://github.com/postmanlabs/openapi-to-postman/pull/825) for openapi-to-postman is merged in.
We will have to wait for the next release of openapi-to-postman to have it in Portman.

**Comment by thim81** [comment]
hi @miriamgreis 

It took a while but openapi-to-postman released a new major version, which we included in the latest [v1.31.0](https://github.com/apideck-libraries/portman/releases/tag/v1.31.0) release of Portman.

Thanks for reporting and being such a great promotor of Portman 🙌 .

Can you validate if your issue now fully resolved?
If it is, feel free to close this item.

**Comment by thim81** [comment]
hi @miriamgreis 

I'm doing a summer cleanup of solved issues. I'm going to close this issue, since it should be resolved with the previous release.
Let me know, if it is still the case, so we can further investigate.
